test/testdrive/kafka-upsert-sources.td

```diff
@@ -659,7 +659,11 @@ ALTER SYSTEM SET enable_envelope_upsert_inline_errors = true
   VALUE FORMAT JSON
   ENVELOPE UPSERT (VALUE DECODING ERRORS = (INLINE));
 
-> SELECT key, value::text, error::text FROM value_decode_error
+# there is now an additional 'error' column that should store the inline decoding errors, if any
+# since this is a json value record the output relation just has a single jsonb column named 'data'
+# though for other types (e.g. avro) the value columns should be present and flattened.   
```

Comment:
I wish there were an easy way to test a decode failure on an avro schema source, but it would require a non-trivial refactoring of the `kafka-ingest` testdrive action to allow it to accept malformed data on an avro topic. Maybe something we can add later (cc @MaterializeInc/testing )
So this just tests on a json upsert source for now

Reply:
Can't you just use `$ kafka-ingest format=text`? I think we do that elsewhere. You don't need to ingest in testdrive with the same format that you declare the source with in Materialize.